### DISCUSSION TITLE
Usage of BeforeStep annotation?

### DISCUSSION THREAD

**Discussion by jcsia87** [getgauge/gauge#2266] [General]
Hi, we have a spec file in the format below, all steps are implemented using Java.

> \# My Spec
> 
> \## My Test Case
> Tags: smoke, regression
>
> \* Step one
> \* Step two
> \* Step x
> \* Step two
> \* Step x

We want a method to run before any `Step two` above executes. Is it possible to do so in Gauge using Java? I see below being referenced in the official Gauge doc but is there a way to add a filter to only execute it before certain steps (in this case `Step two`)?

> @BeforeStep
>     public void BeforeStep() {
>         // Code for before step
>     }

**Comment by sriv** [comment]
Every hook can take a`ExecutionContext` ex: `public void BeforeStep(ExecutionContext ctx)`

This will then allow you to access the current step using [`ExecutionContext.getCurrentStep`](https://javadoc.io/doc/com.thoughtworks.gauge/gauge-java/latest/com/thoughtworks/gauge/ExecutionContext.html#getCurrentStep()), and you can then use any condition based on [`StepDetails.getText`](https://javadoc.io/doc/com.thoughtworks.gauge/gauge-java/latest/com/thoughtworks/gauge/StepDetails.html)
